crates/did-utils/src/methods/peer/method.rs

```diff
@@ -221,18 +221,19 @@ impl DidPeer {
     // ---------------------------------------------------------------------------
 
     /// Expands `did:peer` address into DID document
-    pub fn expand(&self, did: &str) -> Result<DIDDocument, DIDPeerMethodError> {
+    pub fn expand(&self, did: &str) -> Result<DIDDocument, DIDResolutionError> {
         if !did.starts_with("did:peer:") {
-            return Err(DIDPeerMethodError::InvalidPeerDID);
+            return Err(DIDResolutionError::InvalidDid);
         }
 
         match did {
-            s if s.starts_with("did:peer:0") => self.expand_did_peer_0(did),
-            s if s.starts_with("did:peer:2") => self.expand_did_peer_2(did),
-            s if s.starts_with("did:peer:4") => self.expand_did_peer_4(did),
-            _ => Err(DIDPeerMethodError::UnsupportedPeerDIDAlgorithm),
-        }
+        s if s.starts_with("did:peer:0") => self.expand_did_peer_0(did).map_err(Into::into),  // Convert DIDPeerMethodError to DIDResolutionError
+        s if s.starts_with("did:peer:2") => self.expand_did_peer_2(did).map_err(Into::into),
+        s if s.starts_with("did:peer:4") => self.expand_did_peer_4(did).map_err(Into::into),
```

Comment:
I don't think so, it's true a `fully qualified method call` is used in Rust when you need to specify which trait's method to call, usually in the case of multiple traits providing methods with the same name.

But what i have implemented rather is matching against different prefixes of the DID string (`did:peer:0`, `did:peer:2`, `did:peer:4`) and calling different methods (`expand_did_peer_0`, `expand_did_peer_2`, `expand_did_peer_4`).
![Screenshot from 2024-09-26 09-57-54](https://github.com/user-attachments/assets/b2499299-b354-4dea-90c2-82aa2bf2b0df)


So that statement doesn't apply to this scenario

Reply:
@IngridPuppet I think you should look into this statement if I'm wrong

Reply:
> I don't think so, it's true a `fully qualified method call` is used in Rust when you need to specify which trait's method to call, usually in the case of multiple traits providing methods with the same name.
> So that statement doesn't apply to this scenario

Read my comment again. I'm sure that you didn't understand it.